frappe/contacts/doctype/contact/contact.py

```diff
@@ -198,6 +198,29 @@ def contact_query(doctype, txt, searchfield, start, page_len, filters):
 			'link_doctype': link_doctype
 		})
 
+@frappe.whitelist()
+def address_query(links):
+	import json
+
+	links = [{"link_doctype": d.get("link_doctype"), "link_name": d.get("link_name")} for d in json.loads(links)]
+	result = []
+
+	for link in links:
```

Comment:
Check for permissions... since this a whitelisted method, any user can get list of addresses just by passing link doctype and link name.

Reply:
Add a permission check in for loop